### ISSUE TITLE
UITableViewAlertForLayoutOutsideViewHierarchy

### ISSUE THREAD

**Issue by CocoaBob** [SCENEE/FloatingPanel#249]
Hi,

I'm using UITableView in the floating panel, I always get this warning message.

> [TableView] Warning once only: UITableView was told to layout its visible cells and other contents without being in the view hierarchy (the table view or one of its superviews has not been added to a window). This may cause bugs by forcing views inside the table view to load and perform layout without accurate information (e.g. table view bounds, trait collection, layout margins, safe area insets, etc), and will also cause unnecessary performance overhead due to extra layout passes. Make a symbolic breakpoint at UITableViewAlertForLayoutOutsideViewHierarchy to catch this in the debugger and see what caused this to occur, so you can avoid this action altogether if possible, or defer it until the table view has been added to a window.

I figured out that it's caused by this line of code `surfaceView.superview!.layoutIfNeeded()` in FloatingPanel. The solution is very simple, just adding `if surfaceView.window != nil {}` to make sure it's in a view hierarchy before calling `layoutIfNeeded()`.

So it will look like this (I changed the force unwrapping to optional, which is a better practice)
```
if surfaceView.window != nil {
    surfaceView.superview?.layoutIfNeeded()
}
```

Could you add this for me? Or let me create a pull request?

Thanks!

**Comment by SCENEE** [comment]
Sounds great! I confirmed `surfaceView.window` behavior. Please create a pull request. I would like to release it in v1.6.5 so please set `release-1.6.5` to the target branch.

**Comment by SCENEE** [comment]
Would you like me to add this fix by myself?

**Comment by CocoaBob** [comment]
Hi Shin,

Recently I'm on vacation, I think my initial fix suggestion wasn't enough, as I still got breakpoint pause after implementing that fix. I will come back in September, maybe it's too late for your new version😅

Best,

—

Bob

________________________________

**Comment by SCENEE** [comment]
@CocoaBob,
I'm sorry to hurry. I'm ok after you will come back. Instead I'm happy for your contribution so I'm going to merge it in an appropriate version.

**Comment by SCENEE** [comment]
I suppressed UITableViewAlertForLayoutOutsideViewHierarchy error in #263. If you still have a problem, please let me know 😄

**Comment by DiwakarThapa** [comment]
@SCENEE  I think this issue is not fixed yet because on floating panel v1.7.4 it is showing similar warning. Is there a way to remove **UITableViewAlertForLayoutOutsideViewHierarchy** warning from my app ?  Hoping for quick reply thank you.
